app.py

```diff
@@ -18,65 +18,101 @@ def inject_func():
     else:
         return render_template('login.html')
 
 
 
-@app.route('/logged_in', methods=['GET',"POST"])
-def logged_in():
+@app.route('/login_check', methods=["POST"])
+def login_check():
 
-    users_handler = User()
    
     email = request.form.get("inputEmail")
     password = request.form.get("inputPassword")
 
     if users_handler.validate(email , password):
 
-        return redirect(url_for('index'))
+        userid = users_handler.email_userid[email]
+        response = make_response(redirect(url_for('index')))
+        response.set_cookie('user_id' , str(userid))    
+        return response
 
     else:
-        users_handler.register(email,password)
+    
+        flash('Wrong Email or Password. Please try again, or Sign-up!')
+        return render_template('login.html')
 
-        response = make_response(render_template('index.html'))
-        response.set_cookie('email',email)
 
-        return response
+@app.route('/signup', methods=['GET',"POST"] )
+def signup():
+
+    if request.form.get("submit")=='CLICKED':
+
+        email = request.form.get("inputEmail")
+        password = request.form.get("inputPassword")
+        client_name = request.form.get("client_name")
+
+        if not users_handler.old_user(email):
+
+            users_handler.register(client_name,email,password)
+            userid = users_handler.email_userid[email]
+
+            response = make_response(redirect(url_for('index')))
+            response.set_cookie('user_id' , str(userid)) 
+            return response
+    
+        else:
+
+            flash('Login Failed!')
+            return render_template('signup.html')
+    
+    else:   
+
+        return render_template('signup.html')
+    
 
 @app.route('/saved' , methods=["POST"])
 def saved():
     
     input_name = request.form['Name']
     input_number = request.form['Number']
-    username = request.cookies.get('email')
-    phonebook.insert(username,input_name,input_number)
+    userid = request.cookies.get('user_id')
+    userid = int(userid)
+    username = users_handler.userid_username[userid]
+
+    phonebook.insert(userid,username,input_name,input_number)
 
     flash(f'{input_number} for {input_name} has been saved')
 
     return redirect(url_for('index'))
 
 
-@app.route('/table' , methods=["GET","POST"])
+@app.route('/table' , methods=["GET", "POST"])
 def table():
     
-    username = request.cookies.get('email')
+    if request.cookies.get('user_id'):
+        userid = request.cookies.get('user_id')
+        userid = int(userid)
+        contact_list = phonebook.find_book(userid)
 
-    contact_list = [ i for i in phonebook.db if i.get('client_name')==username ]
-
-    return render_template('list.html' , mylist = contact_list )
+        return render_template('list.html' , mylist = contact_list )
+    else:
+        return render_template('login.html')
 
 
-@app.route('/delete' , methods=["GET","POST"])
+@app.route('/delete' , methods=["POST"])
```

Comment:
Ok let's put the contact id in the route url. So you don't have to rely on the submit button's value. Like `/delete/142`. The route will be parametric, e.g. `/delete/:id`.